binarytree/__init__.py

```diff
@@ -451,6 +466,126 @@ def _generate_values(height, multiplier=1):
     return sample(range(count * multiplier), count)
 
 
+def _get_node_with_id(root, index):
```

Comment:
This function could be simplified by re-using some existing code. 

Your new way of assigning node IDs gives us some **really neat properties** to work with: the ID of a node is equal to the index of the node in the tree's [list representation](https://en.wikipedia.org/wiki/Binary_tree#Arrays).

So you can simply convert `root` into list representation by calling a modified version of `_build_list` (let's call this function`_build_list_2` for now) where you put the nodes themselves into the list instead of only their values. Once you have this function, you can use it like this:
```python
def _get_node_with_id(root, index):
    if not isinstance(index, int) or index < 0:
        raise ValueError
    return _build_list_2(_validate_tree(root))[index]
```
Optionally, `_build_list_2 ` can short circuit when it finds the target node for further optimization.